crates/sui-graphql-rpc/src/types/query.rs

```diff
@@ -172,7 +172,11 @@ impl Query {
             )
             .await?;
 
-        DryRunResult::try_from(res).extend()
+        let latest_cp = Checkpoint::query_latest_checkpoint_sequence_number(ctx.data_unchecked())
```

Comment:
This one is a little bit more tricky than the execution case, because dry run is performing a read of the chain.  Does the response from fullnode return a checkpoint that we can use for this, because again, I don't think the latest checkpoint is the correct choice here.